ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorThread.java

```diff
@@ -61,6 +61,13 @@ public abstract class CompactorThread extends Thread implements Configurable {
   protected String hostName;
   protected String runtimeVersion;
 
+  //Time threshold for compactor thread log
+  //In milliseconds:
+  protected Integer MAX_WARN_LOG_TIME = 1200000; //20 min
+
+  protected long checkInterval;
+
+  public enum CompactorThreadType {INITIATOR, WORKER, CLEANER}
```

Comment:
What about introducing an abstract CompactorThreadType getThreadType() method, implement in Cleaner, Worker, Initiator, and use it in doPostLoopActions()? This way CompactorThreadType type parameter would no longer necessary. However, I don't have a strong preference over this, the current approach is also fine.

Reply:
The reason why the enum approached is followed is to keep it consistent with the implementation in test module.